conf/gulp-tasks/template/filenameutils.js

```diff
@@ -0,0 +1,39 @@
+const TemplateType = require('./templatetype');
+
+const fileNames = {
+  [TemplateType.UMD]: 'answerstemplates.compiled.min.js',
+  [TemplateType.IIFE]: 'answerstemplates-iife.compiled.min.js'
+};
+
+/**
+ * Prefixes the given fileName with the locale.
+ * @param {string} fileName
+ * @param {string} locale
+ * @returns {string}
+ */
+function addLocalePrefix (fileName, locale = 'en') {
+  const prefix = (locale === 'en') ? '' : `${locale}-`;
+  return prefix + fileName;
+}
+exports.addLocalePrefix = addLocalePrefix;
+
+/**
+ * Gets the output fileName for the specified templateType and locale.
+ * @param {TemplateType} templateType
+ * @param {string} locale
+ * @returns {string}
+ */
+exports.getFileName = function (templateType, locale) {
+  const fileName = fileNames[templateType];
+  return addLocalePrefix(fileName, locale);
+};
+
+/**
+ * Gets the precompiled templates fileName for the given locale.
+ * @param {string} locale
+ * @returns {string}
+ */
+exports.getPrecompiledFileName = function (locale) {
+  const precompiledFileName = 'answerstemplates.precompiled.min.js';
```

Comment:
Should this use the `fileNames` const at the top of this file?

Reply:
I originally had it like that but it felt a little weird to have a TemplateType.PRECOMPILED since the precompiled template file isn't really a template type it's just an intermediary file. What do you think?

Reply:
Ah, true, yeah since it doesn't persist I can see how it would be misleading if there was a "TemplateType" for it! Better to leave out of that!